executor/inspection_profile.go

```diff
@@ -61,71 +140,162 @@ func (n *metricNode) getName(label string) string {
 	return name
 }
 
-func (n *metricNode) getValue(pb *profileBuilder) (float64, error) {
-	if n.initialized {
-		return n.value, nil
-	}
-	n.initialized = true
-	v, err := pb.getMetricValue(n)
-	if err != nil {
-		return 0, err
-	}
-	if math.IsNaN(v) {
-		n.value = 0
-	} else {
-		n.value = v
+func (n *metricNode) getValue(pb *profileBuilder) (*metricValue, error) {
+	if !n.initialized {
+		n.initialized = true
+		err := n.initializeMetricValue(pb)
+		if err != nil {
+			return nil, err
+		}
 	}
 	return n.value, nil
 }
 
-func (pb *profileBuilder) getMetricValue(n *metricNode) (float64, error) {
-	if n.labelValue != nil {
-		return n.value, nil
+func (n *metricNode) getLabelValue(label string) *metricValue {
+	value, ok := n.labelValue[label]
+	if !ok {
+		value = &metricValue{}
+		n.labelValue[label] = value
 	}
-	n.labelValue = make(map[string]float64)
-	var query string
-	format := "2006-01-02 15:04:05"
-	queryCondition := fmt.Sprintf("where time >= '%v' and time <= '%v'", pb.start.Format(format), pb.end.Format(format))
+	return value
+}
+
+func (n *metricNode) initializeMetricValue(pb *profileBuilder) error {
+	n.labelValue = make(map[string]*metricValue)
+	n.value = &metricValue{}
+	queryCondition := fmt.Sprintf("where time >= '%v' and time <= '%v' and value is not null and value>0",
+		pb.start.Format(dateTimeFormat), pb.end.Format(dateTimeFormat))
 	if n.condition != "" {
 		queryCondition += (" and " + n.condition)
 	}
+
+	queryRows := func(query string, fn func(label string, v float64)) error {
```

Comment:
The code here is really hard to read, please use the normal function.

Reply:
done.